Sources/Talkshoplive/Core/Chat/ChatProvider.swift

```diff
@@ -324,8 +324,12 @@ public class ChatProvider {
                 Config.shared.isDebugMode() ? print("The subscribeError", error.localizedDescription , "Code", error.reason.rawValue) :()
                 if error.reason == .timedOut {
                     self.delegate?.onStatusChange(error: APIClientError.CHAT_TIMEOUT)
-                } else if error.reason.rawValue == 403 {
-                    self.delegate?.onStatusChange(error: APIClientError.PERMISSION_DENIED)
+                } else  if error.reason.rawValue == 403 {
```

Comment:
@mayuri-talkshoplive Can you do optional chaining here?
`error?.reason?.rawValue`?

And what's the `rawValue` when permission is denied? It's better to check for both. If you can!

Reply:
@daman-talkshoplive  optional chaining is not possible here Because from pubnub listener , we are not receiving error as a optional.

Reply:
Here, Raw value is 403, in that as per walmart requested we have to check it my message to give them separate code for token expired and revoked.